theta_dup/include/theta_sketch_dup.h

```diff
@@ -0,0 +1,2209 @@
+   */
+  vector_u64<A> keys_;
+  uint32_t num_keys_;
+  resize_factor rf_;
+  float p_;
+  uint64_t seed_;
+  uint32_t capacity_;
+
+  // for builder
+  update_theta_sketch_dup_alloc(uint8_t lg_cur_size, uint8_t lg_nom_size,
+                                resize_factor rf, float p, uint64_t seed);
+
+  // for deserialize
+  update_theta_sketch_dup_alloc(bool is_empty, uint64_t theta,
+                                uint8_t lg_cur_size, uint8_t lg_nom_size,
+                                vector_u64<A>&& keys, uint32_t num_keys,
+                                resize_factor rf, float p, uint64_t seed);
+
+  void resize();
+  void rebuild();
+
+  // TODO: support union
+  // friend theta_union_alloc<A>;
+  void internal_update(uint64_t hash);
+  void internal_remove(uint64_t hash);
+
+  // TODO: support intersection
+  // friend theta_intersection_alloc<A>;
+  // TODO: support a_not_b
+  // friend theta_a_not_b_alloc<A>;
+  static inline uint32_t get_capacity(uint8_t lg_cur_size, uint8_t lg_nom_size);
+  static inline uint32_t get_stride(uint64_t hash, uint8_t lg_size);
+
+  /**
+   * search hash values, if exists increase the count and return true, otherwise
+   * insert the value and return false
+   * @param hash: the hash value
+   * @param table: the pointer to the hash table
+   * @param table: lg_size of the current hash table
+   */
+  static bool hash_search_or_insert(uint64_t hash,
+                                    std::pair<uint64_t, int64_t>* table,
+                                    uint8_t lg_size);
+  /**
+   * search hash values, if exists decrease the count
+   * if the count==0, remove the element from hash table and return true else
+   * return false
+   * @param hash: the hash value
+   * @param table: the pointer to the hash table
+   * @param table: lg_size of the current hash table
+   */
+  static bool hash_search_or_remove(uint64_t hash,
+                                    std::pair<uint64_t, int64_t>* table,
+                                    uint8_t lg_size);
+  static bool hash_search(uint64_t hash,
+                          const std::pair<uint64_t, int64_t>* table,
+                          uint8_t lg_size);
+
+  friend theta_sketch_dup_alloc<A>;
+  static update_theta_sketch_dup_alloc<A> internal_deserialize(
+      std::istream& is, resize_factor rf, uint8_t lg_cur_size,
+      uint8_t lg_nom_size, uint8_t flags_byte, uint64_t seed);
+  static update_theta_sketch_dup_alloc<A> internal_deserialize(
+      datasketches_pb::ThetaSketchDup& pb, resize_factor rf,
+      uint8_t lg_cur_size, uint8_t lg_nom_size, uint8_t flags_byte,
+      uint64_t seed);
+  static update_theta_sketch_dup_alloc<A> internal_deserialize(
+      const void* bytes, size_t size, resize_factor rf, uint8_t lg_cur_size,
+      uint8_t lg_nom_size, uint8_t flags_byte, uint64_t seed);
+};
+
+// compact sketch
+
+template <typename A>
+class compact_theta_sketch_dup_alloc : public theta_sketch_dup_alloc<A> {
```

Comment:
Add a description of compact sketches and its use case?

Reply:
Updated. See [this commit](https://github.com/googleinterns/cardinality-estimation-with-data-deletions/pull/2/commits/b563ca7370ab2caa3a8ee740d3194db6cf0ae5bb).